backend/hct_mis_api/apps/payment/api/views.py

```diff
@@ -0,0 +1,152 @@
+        business_area = self.get_business_area()
+        queryset = PaymentPlan.objects.filter(business_area=business_area)
+        program_ids = self.request.user.partner.get_program_ids_for_business_area(str(business_area.id))
+        return queryset.filter(
+            status__in=[
+                PaymentPlan.Status.IN_APPROVAL,
+                PaymentPlan.Status.IN_AUTHORIZATION,
+                PaymentPlan.Status.IN_REVIEW,
+            ],
+            program__in=program_ids,
+        )
+
+    @action(
+        detail=False,
+        methods=["post"],
+        url_path="bulk-action",
+        serializer_class=PaymentPlanBulkActionSerializer,
+    )
+    def bulk_action(self, request: Request, *args: Any, **kwargs: Any) -> Response:
+        serializer = self.get_serializer(data=request.data)
+        if serializer.is_valid(raise_exception=True):
+            action_name = serializer.validated_data["action"]
+            comment = serializer.validated_data.get("comment", "")
+            input_data = {"action": action_name, "comment": comment}
+            business_area = self.get_business_area()
+
+            with transaction.atomic():
+                for payment_plan_id_str in serializer.validated_data["ids"]:
+                    self._perform_payment_plan_status_action(
+                        payment_plan_id_str,
+                        input_data,
+                        business_area,
+                        request,
+                    )
+
+        return Response(status=status.HTTP_204_NO_CONTENT)
+
+    def _perform_payment_plan_status_action(
+        self,
+        payment_plan_id_str: str,
+        input_data: dict,
+        business_area: BusinessArea,
+        request: Request,
+    ) -> None:
+        payment_plan_id = decode_id_string(payment_plan_id_str)
+        payment_plan = get_object_or_404(PaymentPlan, id=payment_plan_id)
+
+        if not self.request.user.has_permission(
+            self._get_action_permission(input_data["action"]),
+            business_area,
+            payment_plan.program_id,
+        ):
+            raise PermissionDenied(
+                f"You do not have permission to perform action {input_data['action']} "
+                f"on payment plan with id {payment_plan.unicef_id}."
+            )
+
+        old_payment_plan = copy_model_object(payment_plan)
+        if old_payment_plan.imported_file:
+            old_payment_plan.imported_file = copy_model_object(payment_plan.imported_file)
+        if old_payment_plan.export_file_entitlement:
+            old_payment_plan.export_file_entitlement = copy_model_object(payment_plan.export_file_entitlement)
+        if old_payment_plan.export_file_per_fsp:
+            old_payment_plan.export_file_per_fsp = copy_model_object(payment_plan.export_file_per_fsp)
+
+        payment_plan = PaymentPlanService(payment_plan).execute_update_status_action(
+            input_data=input_data, user=request.user  # type: ignore
```

Comment:
Why it is ignored?

Reply:
Because `request.user` returns `Union[AbstractBaseUser, AnonymousUser]` so either I ignore typing or I would have to change typing, considering also that `AnonymousUser` will never reach this code so either add `if` just for typing or have confusing typing with `AnonymousUser` inside